iceoryx_hoofs/source/posix_wrapper/unix_domain_socket.cpp

```diff
@@ -305,7 +305,9 @@ UnixDomainSocket::timedReceive(const units::Duration& timeout) const noexcept
         {
             return cxx::error<IpcChannelError>(convertErrnoToIpcChannelError(recvCall.get_error().errnum));
         }
-        return cxx::success<std::string>(std::string(message));
+        // Minus one, because we know that the last Byte is a '\0' due to std::string usage
+        return cxx::success<std::string>(
```

Comment:
@elfenpiff Valid point. I opened issue #1373 I propose the following I'll retroactively link the commits with the new issue keeping the change for `UnixDomainSocket` on this branch because it goes hand in hand. I'll open a 2nd PR for `NamedPipe` and `MessageQueue`.

I think you mixed up the numbering: c'tor 5 was used before, but now c'tor 4 is used to not rely on the null-terminated string but use the return value from `recvfrom` as size.